src/Compilers/Test/Utilities/CSharp/TestOptions.cs

```diff
@@ -44,34 +41,52 @@ public static class TestOptions
         public static readonly CSharpCompilationOptions UnsafeDebugDll = DebugDll.WithAllowUnsafe(true);
         public static readonly CSharpCompilationOptions UnsafeDebugExe = DebugExe.WithAllowUnsafe(true);
 
-        public static CSharpParseOptions WithFeature(this CSharpParseOptions options, string feature, string value)
-        {
-            return options.WithFeatures(options.Features.Concat(new[] { new KeyValuePair<string, string>(feature, value) }));
-        }
-
         public static CSharpParseOptions WithStrictFeature(this CSharpParseOptions options)
         {
-            return options.WithFeature("strict", "true");
+            return options.WithFeatures(options.Features.Concat(new[] { new KeyValuePair<string, string>("strict", "true") }));
         }
 
         public static CSharpParseOptions WithLocalFunctionsFeature(this CSharpParseOptions options)
         {
-            return options.WithFeature(MessageID.IDS_FeatureLocalFunctions.RequiredFeature(), "true");
+            return WithExperimental(options, MessageID.IDS_FeatureLocalFunctions);
         }
 
         public static CSharpParseOptions WithRefsFeature(this CSharpParseOptions options)
         {
-            return options.WithFeature(MessageID.IDS_FeatureRefLocalsReturns.RequiredFeature(), "true");
+            return WithExperimental(options, MessageID.IDS_FeatureRefLocalsReturns);
         }
 
         public static CSharpParseOptions WithTuplesFeature(this CSharpParseOptions options)
         {
-            return options.WithFeature("tuples", "true");
+            return WithExperimental(options, MessageID.IDS_FeatureTuples);
```

Comment:
I think we have another file that has similar helpers. Should we make similar changes there?